.circleci/workflows/build-test.yaml

```diff
@@ -1,44 +1,30 @@
+!yamlscript/v0/
+
+requires-checkout =::
+  requires:
+  - checkout
+
+requires-build =::
+  requires:
+  - build
+
+
 jobs:
 - checkout:
     filters:
       tags:
         ignore: /^v\d+\.\d+\.\d+-?.*/
 
-- prettier-check:
-    requires:
-    - checkout
-
-- test-grammar:
-    requires:
-    - checkout
-
-- build:
-    requires:
-    - checkout
-
-- eslint-check:
-    requires:
-    - build
-
-- test-cljslib:
-    requires:
-    - build
-
-- test-integration:
-    requires:
-    - build
+- prettier-check:: requires-checkout
+- test-grammar::   requires-checkout
+- build::          requires-checkout
 
-- test-e2e:
-    requires:
-    - build
-
-- test-e2e-sub-projects:
-    requires:
-    - build
-
-- test-ts-unit:
-    requires:
-    - build
+- eslint-check::          requires-build
+- test-cljslib::          requires-build
+- test-integration::      requires-build
+- test-e2e::              requires-build
+- test-e2e-sub-projects:: requires-build
+- test-ts-unit::          requires-build
```

Comment:
This is sweet! Just a question: Would it be possible to make the required job an argument somehow? Not saying we should do it, just wondering.

Reply:
https://gist.github.com/ingydotnet/1f8b535cc3e157a6835ec21406c0914d#file-gistfile1-txt-L32-L34
shows 2 possible ways to do it differently (both keeping definitions to single lines).

You could predefine a `requires` fn that takes an arg,
or you could predefine nothing and just use YAML flow syntax.

Might be best to just stick with YAML syntax here as nothing very interesting is happening that really requires a function.

Reply:
See 2082ad149